website/addons/base/__init__.py

```diff
@@ -345,6 +346,11 @@ def deep_url(self):
             return url + '/'
 
     @property
+    def source_url(self):
+        extra = self._metadata_cache['extra']
```

Comment:
```extra``` will always exist because it is a property of WaterButler's [BaseMetadata](https://github.com/CenterForOpenScience/waterbutler/blob/develop/waterbutler/core/metadata.py#L13-27) class :+1: